Source/WebKit/NetworkProcess/storage/NetworkStorageManager.cpp

```diff
@@ -244,14 +244,29 @@ static String originFilePath(const String& directory)
     return FileSystem::pathByAppendingComponent(directory, OriginStorageManager::originFileIdentifier());
 }
 
+void NetworkStorageManager::writeOriginToFileIf(const WebCore::ClientOrigin& origin, const Function<bool()>& condition)
+{
+    auto* manager = m_localOriginStorageManagers.get(origin);
+    if (!manager || manager->didWriteOriginToFile())
+        return;
+
+    auto originDirectory = originDirectoryPath(m_path, origin, m_salt);
+    if (originDirectory.isEmpty())
+        return;
+
+    if (!condition())
```

Comment:
Why do we have to do the condition check so late? Why do we bother running all this code above if the condition will return false anyway?

Reply:
Because the previous checks are cheaper... SQLiteStorageArea::isEmpty() may have file operations